test/distributed/test_c10d_nccl.py

```diff
@@ -438,6 +438,32 @@ def test_restart_pg(self):
         with self.assertRaises(ValueError):
             dist.all_reduce(t1)
 
+    @requires_nccl()
+    @skip_but_pass_in_sandcastle_if(not TEST_MULTIGPU, "NCCL test requires 2+ GPUs")
+    def test_cuda_event_cache(self):
```

Comment:
can you make the test name more descriptive or at least add a comment for what the test is testing?

iiuc the point is to test some particular race on thread destruction

Reply:
yes. Let me update it.